### PR TITLE
Add parameter confirm to GCSToSFTPOperator

### PR DESCRIPTION
**Pull request by Nimo139**
If the SFTP server picks up the file immediately Paramiko cannot check the file size. 
The current operator terminates with a fail after the transfer because the file cannot be found.

### CODE REVIEW

**Comment by potiuk** [comment]
I think we will need a (really small) unit test for that one

File: airflow/providers/google/cloud/transfers/gcs_to_sftp.py [hunk 1]
```diff
@@ -84,6 +84,7 @@ class GCSToSFTPOperator(BaseOperator):
     :param gcp_conn_id: (Optional) The connection ID used to connect to Google Cloud.
     :param sftp_conn_id: The sftp connection id. The name or identifier for
         establishing a connection to the SFTP server.
+    :param bool confirm: whether to do a stat() on the file afterwards to confirm the file size
```
Comment (pankajastro) [inline airflow/providers/google/cloud/transfers/gcs_to_sftp.py]:
```suggestion
    :param confirm: whether to do a stat() on the file afterwards to confirm the file size
```

**Comment by eladkal** [comment]
Hey @Nimo139 do you have time to complete this PR?

**Comment by Nimo139** [comment]
Sorry for the late reply, I didn't have time to set up the test environment during the holidays. I have now adjusted the tests for the parameter so that they pass. I have added a very basic test. If you have any better suggestions on how to test it with the mocks, I am glad to hear them.

**Comment by potiuk** [comment]
Should be good enough - now the only thing left is to rebase to latest `main` and fix failing static checks

File: tests/providers/google/cloud/transfers/test_gcs_to_sftp.py [hunk 2]
```diff
@@ -116,7 +116,55 @@ def test_execute_move_single_file(
...
+            task_id=TASK_ID,
+            source_bucket=TEST_BUCKET,
+            source_object=source_object,
+            destination_path=DESTINATION_SFTP,
+            keep_directory_structure=keep_directory_structure,
+            move_object=True,
+            gcp_conn_id=GCP_CONN_ID,
+            sftp_conn_id=SFTP_CONN_ID,
+            impersonation_chain=IMPERSONATION_CHAIN,
+            confirm=confirm
```
Comment (eladkal) [inline tests/providers/google/cloud/transfers/test_gcs_to_sftp.py]:
```suggestion
            confirm=confirm,
```

File: tests/providers/google/cloud/transfers/test_gcs_to_sftp.py [hunk 3]
```diff
@@ -116,7 +116,55 @@ def test_execute_move_single_file(
...
+        gcs_hook_mock.return_value.delete.assert_called_once_with(TEST_BUCKET, source_object)
+
+    @pytest.mark.parametrize(
+        "source_object, target_object, keep_directory_structure, confirm",
+        [
+            ("folder/test_object.txt", "folder/test_object.txt", True, True),
+            ("folder/subfolder/test_object.txt", "folder/subfolder/test_object.txt", True, True),
+            ("folder/test_object.txt", "test_object.txt", False, True),
+            ("folder/subfolder/test_object.txt", "test_object.txt", False, True),
+
```
Comment (eladkal) [inline tests/providers/google/cloud/transfers/test_gcs_to_sftp.py]:
```suggestion
```

**Comment by eladkal** [comment]
I fixed the static tests but there is a failure with one of the unit tests that you need to look into @Nimo139 
`FAILED tests/providers/apache/beam/operators/test_beam.py::TestBeamRunJavaPipelineOperator::test_exec_dataflow_runner - AssertionError: expected call not found.`
https://github.com/apache/airflow/actions/runs/7443638524/job/20248835783?pr=35704#step:5:1100

**Comment by Nimo139** [comment]
Sorry for the force-push, I accidentally updated without clicking the rebase option. I hope you can still use the PR. 
The beam unit test should now pass.

File: airflow/providers/apache/beam/operators/beam.py [hunk 4]
```diff
@@ -522,6 +522,9 @@ def __init__(
         )
         self.jar = jar
         self.job_class = job_class
+        self.pipeline_options.setdefault("labels", {}).update(
```
Comment (eladkal) [inline airflow/providers/apache/beam/operators/beam.py:525]: Is this change related to the PR?
